### PR TITLE
Added AESCryptoService class to encrypt and decrypt access tokens in headers. Fetch constants from applications.procerties file. Fixed affected tests. Added profiles.

### PR DESCRIPTION
**Pull request by SanchitaJain4**

### CODE REVIEW

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Constructor for dependency injection of AESCryptoService.
+     * @param aesCryptoService instance of AESCryptoService
+     */
+    @Autowired
+    private final AESCryptoService aesCryptoService;
+
+    @Autowired
+    public HeaderFilter(AESCryptoService aesCryptoService) {
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:29]: For testing, we should have a no arg constructor as well with static variables we can use. Please add one before this constructor.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+@Component
+public class AESCryptoService {
+    private static SecretKeySpec secretKey;
+
+    /**
+     * Constructor that sets the secret key using the environment property "canvas.secretKey".
+     *
+     * @param env environment instance
+     */
+    @Autowired
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java:27]: We should have a no arg constructor for testing with default variables. Please add.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+     * all requests that come in go through this method.
+     *
+     * @param request servlet request
+     * @param response servlet response
+     * @param chain filter chain
+     * @throws IOException if an I/O error occurs
+     * @throws ServletException if a servlet error occurs
+     */
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:44]: update the name of the method to `executeFilter` or `runFilter`. Current name is too vague.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+        // logic to filter URLs to implement Authz
+        if (httpRequest.getRequestURI().toLowerCase().startsWith("/oauth2response") ||
+            httpRequest.getRequestURI().startsWith("/login")) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        String headerValue = httpRequest.getHeader("Authorization");
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:56]: Let's set "Authorization" as a class string variable. That way we can update it in one location and avoid typos later.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+        // process request based on header value
+        if (StringUtils.isBlank(headerValue)) {
+            httpResponse.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:61]: This is a void method, what are we returning?

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                    public String getHeader(String name) {
+                        if ("Authorization".equals(name)) {
+                            return decryptedToken;
+                        }
+                        return super.getHeader(name);
+                    }
+                };
+                chain.doFilter(modifiedRequest, httpResponse);
+            } catch (Exception e) {
+                e.printStackTrace();
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:78]: Use a logger instead of a print statement

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+                httpResponse.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+                return;
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:80]: This is a void method, what are we returning?

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java [hunk 8]
```diff
@@ -3,16 +3,25 @@
...
 public class OAuthService {
 
     private final CanvasClientService canvasClientService;
+    private String clientSecret;
+    private String clientId;
     //TODO: Add and fetch values from config
-    private final String CLIENT_ID = "10000000000002";
-    private final String CLIENT_SECRET = "xppoeCamrsQVsEdZi5g6ykNE1qEJort6Zdn8ojGTcTBzRO15fR0AEdeeCtqcLymI";
+
+    @Autowired
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:19]: Add a no arg constructor with default variables we can use for testing and as a fall back.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+
+/**
+ * AESCryptoService class that provides methods to encrypt and decrypt Access tokens.
+ */
+@Component
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java:18]: This should be `@Service` since this is business logic

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java [hunk 10]
```diff
@@ -3,16 +3,25 @@
 import com.canvas.exceptions.CanvasAPIException;
 import com.fasterxml.jackson.databind.JsonNode;
 import okhttp3.Response;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
 
 /**
  * This class handles OAuth requests for generation, validation of access tokens
  */
 public class OAuthService {
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:12]: Add `@Service` since this has business logic in the code

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java [hunk 11]
```diff
@@ -3,16 +3,25 @@
...
-    private final String CLIENT_ID = "10000000000002";
-    private final String CLIENT_SECRET = "xppoeCamrsQVsEdZi5g6ykNE1qEJort6Zdn8ojGTcTBzRO15fR0AEdeeCtqcLymI";
+
+    @Autowired
+    public OAuthService(Environment env, CanvasClientService canvasClientService) {
+        this.clientId = env.getProperty("canvas.clientId");
+        this.clientSecret = env.getProperty("canvas.clientSecret");
+        this.canvasClientService = canvasClientService;
+    }
     public OAuthService(CanvasClientService canvasClientService){
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:25]: This is showing as never used in my IDE. What's the purpose of this method?

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java [hunk 12]
```diff
@@ -26,7 +35,7 @@ public OAuthService(CanvasClientService canvasClientService){
      */
     public String fetchAccessTokenResponse(String code, String redirectUri) throws CanvasAPIException {
         try {
-            Response response = canvasClientService.fetchAccessTokenResponse(CLIENT_ID, CLIENT_SECRET, code, redirectUri);
+            Response response = canvasClientService.fetchAccessTokenResponse(clientId, clientSecret, redirectUri, code);
             return parseAccessTokenResponse(response);
 
         } catch (Exception e) {
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:41]: Add a log for exceptions

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+import org.codehaus.plexus.util.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Filter class for header processing.
+ */
+@Component
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:18]: Use `@Service` since this is business logic

File: backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java [hunk 14]
```diff
@@ -33,11 +28,17 @@
 import org.junit.jupiter.params.provider.MethodSource;
 import org.junit.jupiter.params.provider.ValueSource;
 import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java:34]: This is never used, can you delete it please?

File: backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java [hunk 15]
```diff
@@ -33,11 +28,17 @@
 import org.junit.jupiter.params.provider.MethodSource;
 import org.junit.jupiter.params.provider.ValueSource;
 import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java:33]: Why aren't we using this instead import instead `import org.springframework.test.web.servlet.MockMvc;`

File: backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java [hunk 16]
```diff
@@ -79,14 +84,26 @@ class ChromeApiControllerUnitTest {
...
+    @MockBean
+    AESCryptoService aesCryptoService;
+
+    @Autowired
+    Environment env;
+
     ChromeApiController controller;
 
     /**
      * Placeholder for action to take before running tests
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java:96]: This comment is out of date now and should be updated.

File: backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java [hunk 17]
```diff
@@ -441,22 +461,22 @@ void deleteStudentSubmissionFiles_shouldReturnResponseEntityOK() throws Exceptio
 
     private static Stream<Arguments> retrieveStudentSubmissionFiles_shouldThrowIncorrectRequestParamsException_whenAnyParameterIsNull() {
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java:462]: CSV Parameterized tests are cleaner for the code, consider moving a CsvFileSource Parameterized Test instead. Example can be found on line 160-161

File: backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/AESCryptoServiceTest.java [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Before
+    public void init(){
+        env = mock(Environment.class);
+        when(env.getProperty(any()))
+                .thenReturn("someSecret");
+    }
+
+    @Test
+    public void testEncryptAndDecrypt() {
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/AESCryptoServiceTest.java:25]: Please follow the same pattern of Setup, Act, Assert. It's makes it easier to follow your tests

Comment (gaaliciA1990) [inline backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/AESCryptoServiceTest.java:25]: This is not a descriptive test name, what are you testing and expecting? Applies to all test names in this class

File: backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/CanvasClientServiceTest.java [hunk 19]
```diff
@@ -19,20 +19,25 @@
 import org.mockito.Captor;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.env.Environment;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/CanvasClientServiceTest.java:26]: This import is never used, please remove

**Review by gaaliciA1990 — CHANGES_REQUESTED** [review]
Hey @SanchitaJain4 . The overall code looks good and everything is passing our checks. My feedback is lot of framework changes and standard practice changes. Let me know if you have any questions. Thanks!

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+    private static SecretKeySpec secretKey;
+
+    /**
+     * Constructor that sets the secret key using the environment property "canvas.secretKey".
+     *
+     * @param env environment instance
+     */
+    @Autowired
+    public AESCryptoService(Environment env) {
+        this.setKey(env.getProperty("canvas.secretKey"));
```
[the remark this answers is not in the document]
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java:29]: This will never be null because we are reading it from the applications.properties file. Our application shouldn't run without the client secret from the applications.properties file.

File: backend/chromeApiServer/src/main/java/com/canvas/controllers/chromeApiServer/ChromeApiController.java [hunk 21]
```diff
@@ -80,8 +88,10 @@ public ResponseEntity<Submission> retrieveStudentSubmissionFiles(
     ) throws CanvasAPIException, IncorrectRequestParamsException, UserNotAuthorizedException {
         validateGraderRequest(bearerToken, assignmentId, courseId, studentId, userType);
 
-        String userId = canvasClientService.fetchUserId(bearerToken);
-        ExtensionUser graderUser = new ExtensionUser(bearerToken, userId, courseId, assignmentId, studentId, userType);
+        String decryptedBearerToken = request.getHeader("Authorization");
+
+        String userId = canvasClientService.fetchUserId(decryptedBearerToken);
+        ExtensionUser graderUser = new ExtensionUser(decryptedBearerToken, userId, courseId, assignmentId, studentId, userType);
```
[the remark this answers is not in the document]
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/controllers/chromeApiServer/ChromeApiController.java:94]: did not name this variable as part of this change. 
the change here passes the decryptedBearerToken instead of bearerToken

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java [hunk 22]
```diff
@@ -34,23 +33,30 @@
...
     // Logger object
     private static final Logger logger = LoggerFactory.getLogger(CanvasClientService.class);
 
 
     /**
      * Constructor with not arguments
      */
-    public CanvasClientService() {
+    @Autowired
+    public CanvasClientService(Environment env) {
```
[the remark this answers is not in the document]
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java:52]: why? we need the Environment variable loaded into this class for it to work. Hence the change. And we are using Autowired annotation so that when the application starts this is the default constructor that is being used. 
For testing we have the constructor on lines 57 onwards. 
will add comments on what the parameters/variables mean.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 3 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:44]: References
https://www.baeldung.com/spring-boot-add-filter 
https://www.tutorialspoint.com/spring_boot/spring_boot_servlet_filter.htm
https://www.javadevjournal.com/spring-boot/spring-boot-add-filter/
doFilter is a common pattern in all springboot applications. let's stick to it.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 4 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:56]: will do

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 5 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:61]: This prevents the request from going down the chain to the ChromeApiController and instead returns the response back to the user.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 6 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:78]: will do

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 7 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:80]: same comment as above

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java (hunk 12 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:41]: will do

File: backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java (hunk 17 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/test/java/com/canvas/controllers/chromeApiServer/ChromeApiControllerUnitTest.java:462]: this test wasn't added as part of this PR.

File: backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/AESCryptoServiceTest.java (hunk 18 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/AESCryptoServiceTest.java:25]: will do

Comment (SanchitaJain4) [reply backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/AESCryptoServiceTest.java:25]: wil change the test name

File: backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/CanvasClientServiceTest.java (hunk 19 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/CanvasClientServiceTest.java:26]: will do

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     * Helper method to set the secret key.
+     *
+     * @param myKey key to be set as secret key
+     */
+    private void setKey(String myKey) {
+        MessageDigest sha = null;
+        try {
+            byte[] key = myKey.getBytes("UTF-8");
```
[the remark this answers is not in the document]
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java:40]: will do

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+     * @param secret secret to use for encryption
+     * @return encrypted string
+     * @throws CanvasAPIException if error occurs during encryption
+     */
+    public String encrypt(String strToEncrypt, String secret) throws CanvasAPIException {
+        try {
+            //this.setKey(secret);
+            Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
+            cipher.init(Cipher.ENCRYPT_MODE, secretKey);
+            return Base64.getEncoder().encodeToString(cipher.doFinal(strToEncrypt.getBytes("UTF-8")));
```
[the remark this answers is not in the document]
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java:63]: will do

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java (hunk 8 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:19]: good point, will do

Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:19]: I tried to add a no arg constructor but I don't understand how I can create a no arg constructor As I do not want to assign a hard coded value/ a null or empty value. Will address in a future PR. We have coverage for this so this should work for now.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java (hunk 11 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/OAuthService.java:25]: Removed it.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 1 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:29]: Same as [this|https://github.com/Canvas-Code-Capstone/Canvas-Code/pull/57/files/9bcdee2284a82ce7969f0d6edeb5515e45b6dd5c#r1113552167] comment. Will address in a future PR.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java (hunk 9 above)
Comment (gaaliciA1990) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/AESCryptoService.java:20]: This is still saying `@Component`. Please update to `@Service`

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java [hunk 25]
```diff
@@ -43,14 +42,14 @@ public class CanvasClientService {
     private static final Logger logger = LoggerFactory.getLogger(CanvasClientService.class);
 
 
-    /**
-     * Constructor with not arguments
-     */
-    public CanvasClientService() {
+    @Autowired
+    protected CanvasClientService(Environment env) {
+        this.canvasUrl = env.getProperty("canvas.url");
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java:47]: This is intended to be a no-arg constructor that just has the okHttypClient. Undo the changes and leave as is, since your other constructor has the args

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java [hunk 26]
```diff
@@ -43,14 +42,14 @@ public class CanvasClientService {
...
         this.okHttpClient = new OkHttpClient();
     }
 
-    protected CanvasClientService(OkHttpClient okHttpClient) {
+    protected CanvasClientService(String canvasUrl, OkHttpClient okHttpClient) {
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java:51]: Still missing `@Autowired` before the method declaration

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java [hunk 27]
```diff
@@ -43,14 +42,14 @@ public class CanvasClientService {
...
+        this.canvasUrl = canvasUrl;
```
Comment (gaaliciA1990) [inline backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java:52]: I like that we are using a constant, however this will affect the usability of our app having it referenced the http://10.242.15.70:3000 url, since we don't have a database to fetch files. It makes sense to have this for the Oauth, but we will want to have the others reference either https://seattleu.instructure.com/api/v1 or https://canvas.instructure.com/api/v1 for demoing. Can you update that in this file, since all of them have been updated to use the local canvas. Let me know if that doesn't make sense

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 3 above)
Comment (gaaliciA1990) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:50]: Ahh, this is overriding the existing function, got it

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 5 above)
Comment (gaaliciA1990) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:67]: You're using an if else statement, so it should only go into the else statement if line 59 isn't matched. HttpResponse handles returning the response to the user so the return doesn't do anything. My IDE is graying it out because it's not needed.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java (hunk 7 above)
Comment (gaaliciA1990) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/HeaderFilter.java:86]: Same as above, httpResponse handles returns should handle that

**Review by gaaliciA1990 — CHANGES_REQUESTED** [review]
Please address comments :)

**Comment by gaaliciA1990** [comment]
@SanchitaJain4 please address the comments in this PR by EOW. Thanks!

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java (hunk 27 above)
Comment (gaaliciA1990) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java:52]: consider making this the `canvas.oauth.url` and leaving everything else as the original urls we were using.

**Comment by gaaliciA1990** [comment]
Please add the redirect url to the frontend code, popup.jsx. Ensure the redirect works when running the sprint-boot server (should be able to select in the maven side bar) and building the webpack `npm run build`

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java (hunk 25 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java:47]: This cannot be a no-arg constructor. env is a mandatory parameter required for starting the application. Since we have coverage this shouldn't be a problem.

File: backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java (hunk 26 above)
Comment (SanchitaJain4) [reply backend/chromeApiServer/src/main/java/com/canvas/service/helperServices/CanvasClientService.java:51]: Since there can only be one @autowired constructor this cannot be @autowired too.

**Review by gaaliciA1990 — APPROVED** [review]
Everything looks good, just fix the merge issues and it's ready push
